### DISCUSSION TITLE
Use trained model to predict new videos

### DISCUSSION THREAD

**Discussion by SkyYu3321** [talmolab/sleap#1684] [Help!]
I wonder how to use a trained model to predict new videos OR do I have to go through manual labeling every time?

**Comment by Marielamend** [comment]
Hello @SkyYu3321!,

1) To use a trained model to make predictions on new videos and utilize old data, you want to group the videos into the same **SLEAP labels project**.
You can add the videos under **File** --> **Add Videos**

2) You can merge two separate projects you have already labeled separately into a combined project by going to 
**File** --> **Merge Into Project**... and select the other SLP file to merge into your current project.

When merging, it is recommended to save the project under a new name selecting **Save As**.

![image](https://github.com/talmolab/sleap/assets/120222245/a56a5fbc-89cd-4692-bd92-cfaf48f63527)

3) In regards to the trained model:

> When you have other models in the `models/` subfolder in the same directory as your SLP file, they'll appear in the dropdown in the **Run Training** window, with a "**[Trained]**" prefix next to each model’s training_config.json.
> 
> You can use this feature to quickly reuse the settings you had configured previously, but you'll be training a completely new model.
> 
> When you select "**Use Trained Model**", all the fields will be grayed out no training will occur for that model and SLEAP will just use the previously trained model.
> 
> If you select "**Resume Training**", however, what SLEAP does is to train a new model, but it initializes it with the previous model weights. This is what is known as "transfer learning", or essentially reusing part of what it learned in the previous run as a starting point. This can help when you have a model trained on a different but related dataset that can bootstrap your model optimization. It does not, however, have any guarantee that it'll "remember" the old data nor work very well on that data. To train a model that works well on both datasets, you have to include both datasets in the same SLEAP project.

Best,
Mariela
